### PR TITLE
1039 Add historical bloods stage

### PR DESCRIPTION
**Pull request by foster999**
## Pre-review checklist

I have ensured that:

- [ ] Code runs successfully in the development environment.
- [ ] New code is tested to prove that it works. If not, justify reasoning.
- [ ] New code is documented using [type hinting](https://realpython.com/lessons/type-hinting/) and following the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) docstring format.
- [ ] New code follows the project naming standards.

## Post-review checklist

As I merge these changes, I have:

- [ ] Updated the related Jira ticket, including approximate time spent (including review).
- [ ] Moved the related Jira ticket to Done.

### CODE REVIEW

File: cishouseholds/pipeline/historic_blood_ETL.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from cishouseholds.pipeline.load import update_table
+from cishouseholds.pipeline.pipeline_stages import register_pipeline_stage
+from cishouseholds.pipeline.timestamp_map import blood_datetime_map
+from cishouseholds.pipeline.validation_schema import historic_blood_validation_schema
+from cishouseholds.pyspark_utils import convert_cerberus_schema_to_pyspark
+from cishouseholds.pyspark_utils import get_or_create_spark_session
+from cishouseholds.validate import validate_and_filter
+
+
+@register_pipeline_stage("blood_delta_ETL")
```
Comment (foster999) [inline cishouseholds/pipeline/historic_blood_ETL.py:22]: Needs a distinct name, as this matches the original blood one

File: cishouseholds/pipeline/timestamp_map.py [hunk 2]
```diff
@@ -20,3 +20,8 @@
     "yyyy-MM-dd HH:mm:ss": ["blood_sample_collected_datetime"],
     "yyyy-MM-dd": ["blood_sample_arrayed_date", "antibody_test_result_recorded_date", "blood_sample_received_date"],
 }
+
+historic_blood_datetime_map = {
+    "yyyy-MM-dd HH:mm:ss": ["arrayed_ox_date"],
```
Comment (foster999) [inline cishouseholds/pipeline/timestamp_map.py:25]: These are using the stata names, so need updating to use those renamed to above to correctly parse to timestamp

File: tests/pipeline/test_historic_blood.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+def test_historic_blood_delta_ETL_df(regression_test_df, historic_blood_delta_ETL_output):
+    regression_test_df(
+        historic_blood_delta_ETL_output.drop("csv_filename"), "ons_id", "processed_historic_blood"
+    )  # removes filename column to account for variation in filename caused by regression
+
+
+@pytest.mark.integration
+def test_historic_blood_delta_ETL_schema(regression_test_df_schema, historic_blood_delta_ETL_output):
+    regression_test_df_schema(
+        historic_blood_delta_ETL_output.drop("csv_filename"), "processed_historic_blood"
```
Comment (foster999) [inline tests/pipeline/test_historic_blood.py:42]: Can delete this drop for the schema, as the schema doesn't vary between test runs
```suggestion
        historic_blood_delta_ETL_output, "processed_historic_blood"
```

File: tests/pipeline/test_historic_blood.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        transform_blood_delta,
+    )
+
+    return processed_df
+
+
+@pytest.mark.integration
+def test_historic_blood_delta_ETL_df(regression_test_df, historic_blood_delta_ETL_output):
+    regression_test_df(
+        historic_blood_delta_ETL_output.drop("csv_filename"), "ons_id", "processed_historic_blood"
```
Comment (foster999) [inline tests/pipeline/test_historic_blood.py:35]: I think this should drop the new source file column and use the new ID: 
```suggestion
        historic_blood_delta_ETL_output.drop("historic_blood_test_source_file"), "blood_sample_barcode", "processed_historic_blood"
```

File: cishouseholds/pipeline/timestamp_map.py (hunk 2 above)
Comment (slamothe99) [reply cishouseholds/pipeline/timestamp_map.py]: modified to pipeline names

**Comment by foster999** [comment]
Thanks @slamothe99, I'm waiting on seeing the real file later today just to check the column names here match. Otherwise looks good now

**Pull request merged**